### DISCUSSION TITLE
Roda ERB active menu helper

### DISCUSSION THREAD

**Discussion by teamschlaflos** [jeremyevans/roda#349] [General]
I'd like to create a menu helper function for my ERB templates. My initial thought was to have a menu variable for each route, but then I thought maybe I'd like to avoid to pollute roda route code with that information, but rather keep the decision in an ERB template:

```erb
 <a class="nav-link <%= active_if "/section/item1" %>">Item 1</a>
 <a class="nav-link <%= active_if "/section/item2" %>">Item 2</a>
```

The issue I'm having is that I'm unable to access `r` in the function:

```ruby
class App
  # ..
  def active_if(path)
    r.matched_path.start_with?(path) ? 'active' ? ''
  end
end
```

Can I somehow access `r` and does anyone have a nice way to handle menus or submenus in an app?

**Comment by adam12** [comment]
Use `request` in replace of `r`. I believe there's also a plugin that will do this for you.

**Comment by teamschlaflos** [comment]
Thanks, that worked! I couldn't make out the correct plugin ([plugin list](https://roda.jeremyevans.net/rdoc/classes/Roda/RodaPlugins.html)).

Here is the updated version which works for me, without the plugin:

```erb
 <a class="nav-link <%= active_if "/section/item1" %>">Item 1</a>
 <a class="nav-link <%= active_if "/section/item2" %>">Item 2</a>
```

```ruby
class App
  # ..

  def active_if(path)
    if request.matched_path.start_with?(path)
      'active'
    else
      ''
    end
  end
end
```

**Reply by adam12** [reply]
Glad it worked. [This is the plugin](https://roda.jeremyevans.net/rdoc/classes/Roda/RodaPlugins/R.html) I was thinking of.
